src/de.rs

```diff
@@ -147,10 +146,6 @@ enum ErrorKind {
         found: &'static str,
     },
 
-    /// An array was decoded but the types inside of it were mixed, which is
-    /// disallowed by TOML.
-    MixedArrayType,
```

Comment:
Could this be left in temporarily to avoid a breaking change, but we can file an issue to delete it on the next breaking change?

Reply:
Yeah, you're right. I'm on it.

Reply:
@alexcrichton Also, what could be the best approach to handle `dead_code` warning, since `MixedArrayType` isn't used anywhere. I don't think it is a good idea to put `#[allow(dead_code)]`, but I can't come up with another solution..

Reply:
This enum is private so it isn't breaking to remove variants.

Reply:
Er my mistake, I didn't realize this was private. Yes it's private so it's fine to delete.